container/monitor.go

```diff
@@ -249,13 +241,6 @@ func (m *containerMonitor) resetMonitor(successful bool) {
 		// the process.  We will build up by multiplying the increment by 2
 		m.timeIncrement *= 2
 	}
-
-	// the container exited successfully so we need to reset the failure counter
```

Comment:
`failureCount` is only used in 
`func (m *containerMonitor) shouldRestart(exitCode int) bool` for calculating the Restart Condition, I have merged `container.ShouldRestart` and `containerMonitor.shouldRestart` as @cpuguy83 suggested, and `failureCount` can be replaced with `container.RestartCount`

We don't need it, hence removed.